src/keys.rs

```diff
@@ -35,7 +35,9 @@ impl SpendingKey {
         let sk = SpendingKey(sk);
         // If ask = 0, discard this key.
         let ask = SpendAuthorizingKey::derive_inner(&sk);
```

Comment:
This is not doing the adjustment to make the ỹ bit 0. We can't do
```
        let ask = SpendAuthorizingKey::from(&sk);
```
because the assertion on line 63 might fail. The simplest way to fix this is to use `from` but remove the assertion on line 63. (The possible negation will not change ask if it is 0.)